src/cli.c

```diff
@@ -0,0 +1,80 @@
+#include "cli.h"
+#include "debug.h"
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <limits.h>
+#include <getopt.h>
+#include <stdio.h>
+
+#define MAX_HZ (1000000L)
+
+static const char* usage_str = "cretro [-d <0..5>] <HERTZ> <ROM>";
+
+__attribute__((__const__)) config_t cli_config_default(void) {
+    config_t conf = {
+        .debug = 0,
+        .us_delay = -1,
+        .rom = "",
+    };
+
+    return conf;
+}
+
+int cli_config_handle(config_t* const conf, int argc, char **argv) {
+    int c;
+
+    // parse all options first
+    while ((c = getopt(argc, argv, "d:")) != -1) {
+        switch (c) {
+            case 'd':
+                conf->debug = (int) strtol(optarg, NULL, 10);
+                debug_init(conf);
+                break;
+            default:
+                fprintf(stderr, "%s\n", usage_str);
```

Comment:
Maybe this should you `LOG_*`

Reply:
`LOG_*` is prefixed with the logging level and the current date, I don't think that this would look right
```
FATAL – Thu 13 10 2022 14:55:25 – USAGE: cretro [-d <0..5>] <FREQUENCY> <ROM>
```

Reply:
Right, I didn't think of that, then nvm